core/templates/components/button-directives/hint-and-solution-buttons.component.ts

```diff
@@ -72,11 +72,8 @@ export class HintAndSolutionButtonsComponent implements OnInit, OnDestroy {
       this.playerPositionService.onNewCardOpened.subscribe(
         (newCard: StateCard) => {
           this.displayedCard = newCard;
-          const solution = newCard.getSolution();
-          if (solution) {
-            this.hintsAndSolutionManagerService.reset(
-              newCard.getHints(), solution);
-          }
+          this.hintsAndSolutionManagerService.reset(
```

Comment:
I don't think this should compile. Reset takes `Solution` as the second argument, while `getSolution` returns `Solution | null`